app/presenters/track.rb

```diff
@@ -0,0 +1,39 @@
+module ExercismWeb
+  module Presenters
+    class Track
+
+      def initialize(track_id)
+        @track_id = track_id
+        @track = Trackler.tracks[track_id]
+      end
+
+      def method_missing(method, *args)
+        @track.send(method, *args)
+      end
+
+      def docs
+        track_docs = @track.docs("/api/v1/tracks/%s/images/docs/img" % @track_id)
+
+        track_docs.each_pair do |topic_name, topic_content|
+          track_docs[topic_name] = if topic_content.present?
```

Comment:
We are reading twice from the file for each `topic_name` here (in case the topic_content is not present). May I suggest something like:

```ruby
track_docs[topic_name] = if topic_content.present?
  topic_content
else
  default_topic_content(topic_name)
end

private

def default_topic_content(topic_name)
  filepath = "./x/docs/md/track/#{topic_name.upcase}.md"

  return '' unless File.exist?(filepath)
  File.read(filepath)
```